docs/howto/krustlet-on-lke.md

```diff
@@ -0,0 +1,281 @@
+
+Please set the value of `PUBLIC_KEY` to point to the file containing the public
+key you will use to authenticate the Linode:
+
+You will also need to identify the Linode user. You can do this with the following
+command. Set the value of `USER` to the user you wish to use:
+
+```console
+$ linode-cli show-users
+```
+
+You can create a Linode (VM) with the following command to create a Linode in the
+`us-west` region using a `g6-standard-1` type with a randomly-generated root
+password and your SSH public key:
+
+```console
+$ LABEL=[[YOUR-LABEL]]
+$ TYPE="g6-standard-1"
+$ REGION="us-west"
+$ IMAGE="linode/debian10"
+$ PASSWORD="$(</dev/urandom tr -dc a-z0-9 | head -c "${1:-32}";echo;)"
+
+$ linode-cli linodes create \
+  --type=${TYPE} \
+  --region=${REGION} \
+  --image=${IMAGE} \
+  --root_pass=${PASSWORD} \
+  --authorized_keys="$(cat ${PUBLIC_KEY})" \
+  --booted=true \
+  --label=${LABEL} \
+  --authorized_users=${USER}
+```
+
+You may then grab the Linode's `ID` and public IP `IP` using:
+
+```console
+$ LINODE_ID=$(linode-cli linodes list \
+  --label=${LABEL} \
+  --format=id \
+  --no-headers \
+  --text) && echo ${LINODE_ID}
+
+$ IP=$(linode-cli linodes list \
+  --label=${LABEL} \
+  --format=ipv4 \
+  --no-headers \
+  --text) && echo ${IP}
+```
+
+## Step 2: Get a bootstrap config for your Krustlet node
+
+Krustlet requires a bootstrap token and config the first time it runs. Follow
+the guide [here](bootstrapping.md), setting the `CONFIG_DIR` variable to `./`,
+to generate a bootstrap config and then return to this document. If you already
+have a kubeconfig available that you generated through another process, you can
+proceed to the next step. However, the credentials Krustlet uses must be part of
+the `system:nodes` group in order for things to function properly.
+
+NOTE: You may be wondering why you can't run this on the VM you just
+provisioned. We need access to the Kubernetes API in order to create the
+bootstrap token, so the script used to generate the bootstrap config needs to be
+run on a machine with the proper Kubernetes credentials.
+
+## Step 3: Copy bootstrap config to VM
+
+The first thing we'll need to do is copy up the assets we generated in steps 1
+and 2. Copy them to the Linode by typing:
+
+```console
+$ scp \
+  -i ${PRIVATE_KEY} \
+  ${HOME}/.krustlet/config/bootstrap.conf \
+  root@${IP}:.
+```
+
+> **NOTE** You'll need to set `PRIVATE_KEY` to the path to the private key
+corresponding to the SSH key pair that you used to create the Linode.
+
+We can then SSH into the Linode by typing:
+
+```console
+$ ssh \
+  -i ${PRIVATE_KEY} \
+  root@${IP}
+```
+
+## Step 4: Install and configure Krustlet
+
+Install the latest release of krustlet following [the install
+guide](../intro/install.md).
+
+There are two flavors of Krustlet (`krustlet-wasi` and `krustlet-wascc`), let's
+use the first:
+
+```console
+$ KUBECONFIG=${PWD}/kubeconfig krustlet-wasi \
+--hostname="krustlet" \
+--node-ip=${IP} \
```

Comment:
I am not entirely sure based on step 1, but is this the public IP of the node? This means that your logs and exec API would be exposed to the public. This is obviously fine for testing, but would be a security problem otherwise, so I'd make sure to call that out

Reply:
Added a caveat: https://github.com/deislabs/krustlet/pull/497/commits/7be22e5b22658b8d424e12a52548564c1ee9ffbb